nuget/include/ShaderGenerator.h

```diff
@@ -27,70 +24,59 @@ namespace ShaderGenerator
     struct ShaderBlockInfo
     {
       uint64_t CompressedOffset = 0ull;
+      uint64_t CompressedLength = 0ull;
       uint32_t ShaderCount = 0u;
     };
 
     struct ShaderBlock
     {
       uint64_t Key;
       std::unordered_map<uint64_t, uint64_t> ShaderOffsets;
-      winrt::Windows::Storage::Streams::InMemoryRandomAccessStream Block;
+      std::stringstream Block;
     };
-#pragma endregion
 
-#pragma region Helper methods
-    template <typename TAsync>
-    static void AwaitOperation(TAsync const& async)
+    struct decompressor_handle_traits
     {
-      auto event = winrt::handle(CreateEvent(nullptr, true, false, nullptr));
-      async.Completed([&](auto&&...)
-        {
-          SetEvent(event.get());
-        });
+      using type = DECOMPRESSOR_HANDLE;
 
-      WaitForSingleObject(event.get(), INFINITE);
-    }
+      static void close(type value) noexcept
+      {
+        CloseDecompressor(value);
+      }
 
-    template <typename TAsync>
-    static auto AwaitResults(TAsync const& async)
-    {
-      AwaitOperation(async);
-      return async.GetResults();
-    }
+      static type invalid() noexcept
+      {
+        return reinterpret_cast<type>(-1);
```

Comment:
We could also write:

return static_cast\<type\>(INVALID_HANDLE_VALUE);

Reply:
I might leave this, since I just copied it over from winrt/base.h from another handle trait there